stripe/src/main/java/com/stripe/android/model/ShippingMethod.java

```diff
@@ -0,0 +1,109 @@
+package com.stripe.android.model;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.json.JSONObject;
+
+import java.util.Currency;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.stripe.android.model.StripeJsonUtils.putLongIfNotNull;
+import static com.stripe.android.model.StripeJsonUtils.putStringIfNotNull;
+
+/**
+ * Model representing a shipping method in the Android SDK
+ */
+public class ShippingMethod extends StripeJsonModel {
+
+    private static final String FIELD_AMOUNT = "amount";
+    /*ISO Currency Code*/
+    private static final String FIELD_CURRENCY_CODE = "currency_code";
+    private static final String FIELD_DETAIL = "detail";
+    private static final String FIELD_IDENTIFIER = "identifier";
+    private static final String FIELD_LABEL = "label";
+
+    private long mAmount;
+    private String mCurrencyCode;
+    private @Nullable String mDetail;
+    private String mLabel;
+    private String mIdentifier;
+
+    public ShippingMethod(String label, String identifier, long amount, String currencyCode) {
```

Comment:
Probably should have this call the other overload with a `null` value for `detail` to avoid duplication. Also, javadoc4public